### DISCUSSION TITLE
Ellies Tutorial Devblog!

### DISCUSSION THREAD

**Discussion by Ellie-x3** [MonoGame/MonoGame#8271] [Showcase]
This is the start of my devblog here on Github Discussions moving from the MonoGames discord forum page.

The first 3 parts have been added already! feel free to post here with any questions or reach out to me on discord. tag: ellie.x3 

 https://elliemakesgames.xyz/

Edit: new domain https://elliemakesgames.net/

**Comment by Ellie-x3** [comment]
Created a new link! https://elliemakesgames.net/ the xyz domain still works though!

**Comment by Bletraut** [comment]
I looked at your blog and wanted to give some tips about the Input class. In part 3 in the **Update** method, there is no need to do extra work and save the state of the keys into an array; this is a useless functionality because the keyboard state already contains all the states of the keys.
The fixed version
```
using System;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Input;

namespace PixelPlatformer.Util;

public static class Input {
    private static KeyboardState _currentKeyboard;
    private static KeyboardState _previousKeyboard;

    public static void Update(GameTime gameTime){
        _previousKeyboard = _currentKeyboard;
        _currentKeyboard = Keyboard.GetState();
    }

    public static bool IsKeyPressed(Keys key) => _currentKeyboard.IsKeyDown(key) && !_previousKeyboard.IsKeyDown(key);

    public static bool IsKeyDown(Keys key) => _currentKeyboard.IsKeyDown(key);
}
```

**Reply by Ellie-x3** [reply]
Thats fair and a better solution - but keep in mind this tutorial isnt 100% prepared and im building the game as I do parts its not already done. so there is going to be things that are changed / im probably not going to have the super optimal best code and best looking code for something and things are subject to change. But I appreciate the tip . im just saying this since theres probably going to be other situations where there is a more clever solution to something that in the moment I just didnt think of to do.

**Reply by Bletraut** [reply]
Yes, there will undoubtedly be such places. And it’s better not to focus on the perfection of the code. But in this case, there is a more effective solution (and i have it) that will help improve your work. Perhaps, having this example, your future code will become better. I wish you success with your game. 😄
